libkbfs/journal_md_ops.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+)
+
+// A journalMDOps is an implementation of MDOps that delegates to a
+// TLF's mdJournal, if one exists. Specifically, it intercepts put
+// calls to write to the journal instead of the MDServer, where
+// something else is presumably flushing the journal to the MDServer.
+//
+// It then intercepts get calls to provide a combined view of the MDs
+// from the journal and the server when the journal is
+// non-empty. Specifically, if rev is the earliest revision in the
+// journal, and BID is the branch ID of the journal (which can only
+// have one), then any requests for revisions >= rev on BID will be
+// served from the journal instead of the server. If BID is empty,
+// i.e. the journal is holding merged revisions, then this means that
+// all merged revisions on the server from rev are hidden.
+//
+// TODO: This makes server updates meaningless for revisions >=
+// rev. Fix this.
+type journalMDOps struct {
+	MDOps
+	jServer  *JournalServer
+	mdServer MDServer
```

Comment:
As always I'd prefer if this could be changeable dynamically by keeping a `Config` here instead, unless there's a good reason not to.